<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<!-- Manual page created with latex2man on Sat Apr 23 05:46:00 EDT 2011 --
-- Author of latex2man: Juergen.Vollmer@informatik-vollmer.de --
-- NOTE: This file is generated, DO NOT EDIT. -->
<html>
<head><title>JELLYFISH</title></head>
<body bgcolor="white">
<h1 align=center>
Jellyfish: A fast k-mer counter 
</h1>
<h4 align=center>G. Marcais </h4>
<h4 align=center>2010/10/1</h4>
<h4 align=center>Version 1.1</h4>
<p>
<tt>Jellyfish</tt>
is a software to count k-mers in DNA sequences. 
<p>
<h3>Table of Contents</h3>
<ul>
<li><a href="#section_1">Synopsis </a></li>
<li><a href="#section_2">Description </a></li>
<li><a href="#section_3">Options </a>
<ul>
<li><a href="#section_4">count </a></li>
<li><a href="#section_5">histo </a></li>
<li><a href="#section_6">dump </a></li>
<li><a href="#section_7">stats </a></li>
<li><a href="#section_8">merge </a></li>
<li><a href="#section_9">cite </a></li>
</ul>
<li><a href="#section_10">Version </a></li>
<li><a href="#section_11">Bugs </a></li>
<li><a href="#section_12">Copyright &amp; License </a></li>
<li><a href="#section_13">Authors </a></li>
</ul>
<p>
<h2><a name="section_1">Synopsis</a></h2>

<tt>jellyfish count</tt>
[<b>-o</b><i>prefix</i>]
[<b>-m</b><i>merlength</i>]
[<b>-t</b><i>threads</i>]
[<b>-s</b><i>hashsize</i>]
[<b>--both-strands</b>]
<i>fasta</i>
[<i>fasta ...</i>]
<br>
<tt>jellyfish merge</tt>
<i>hash1</i>
<i>hash2</i>
...<br>
<tt>jellyfish dump</tt>
<i>hash</i>
<br>
<tt>jellyfish stats</tt>
<i>hash</i>
<br>
<tt>jellyfish histo</tt>
[<b>-h</b><i>high</i>]
[<b>-l</b><i>low</i>]
[<b>-i</b><i>increment</i>]
<i>hash</i>
<br>
<tt>jellyfish query</tt>
<i>hash</i>
<p>
<h2><a name="section_2">Description</a></h2>

<p>
<tt>Jellyfish</tt>
is a k-mer counter based on a multi-threaded hash 
table implementation. 
<p>
To count k-mers, use a command like: 
<p>
<pre>
jellyfish count -m 22 -o output -c 3 -s 10000000 -t 32 input.fasta
</pre>
<p>
This will count the the 22-mers in species.fasta with 32 threads. The 
counter field in the hash uses only 3 bits and the hash has at least 
10 million entries. Let the size of the table be s=2^l and the max 
reprobe value is less than 2^r, then the memory usage per entry in 
the hash is (in bits, not bytes) 2k-l+r+1. 
<p>
To save space, the hash table supports variable length counter, i.e. a 
k-mer occurring only a few times will use a small counter, a k-mer 
occurring many times will used multiple entries in the hash. The 
<b>-c</b>
specify the length of the small counter. The tradeoff is: a 
low value will save space per entry in the hash but will increase the 
number of entries used, hence maybe requiring a larger hash. In 
practice, use a value for <b>-c</b>
so that most of you k-mers 
require only 1 entry. For example, to count k-mers in a genome, 
where most of the sequence is unique, use <b>-c</b><i>1</i>
or 
<b>-c</b><i>2</i>.
For sequencing reads, use a value for 
<b>-c</b>
large enough to counts up to twice the coverage. 
<p>
When the orientation of the sequences in the input fasta file is not 
known, e.g. in sequencing reads, using <b>--both-strands</b>
(<b>-C</b>)
makes the most sense. 
<p>
The following subcommand are used to look at the result: histo, dump, stats. 
<p>
<h2><a name="section_3">Options</a></h2>

<h4><a name="section_4">count</a></h4>

<p>
Count k-mers or qmers in fasta or fastq files 
<p>
Usage: jellyfish count [OPTIONS]... [file.f[aq]]... 
<p>
<dl compact>
<dt><b>-h</b>,<b>--help</b>
</dt>
<dd> Print help and exit 
</dd>
<dt><b>--full-help</b>
</dt>
<dd> Print help, including hidden options, and exit 
</dd>
<dt><b>-V</b>,<b>--version</b>
</dt>
<dd> Print version and exit 
</dd>
<dt><b>-m</b>,
<b>--mer-len</b><i>=INT</i>
</dt>
<dd> Length of mer (mandatory) 
</dd>
<dt><b>-s</b>,
<b>--size</b><i>=LONG</i>
</dt>
<dd> Hash size (mandatory) 
</dd>
<dt><b>-t</b>,
<b>--threads</b><i>=INT</i>
</dt>
<dd> Number of threads (default=1) 
</dd>
<dt><b>-o</b>,
<b>--output</b><i>=STRING</i>
</dt>
<dd> Output prefix (default=mer_counts) 
</dd>
<dt><b>-c</b>,
<b>--counter-len</b><i>=Length</i>
</dt>
<dd> in bits 
Length of counting field (default=7) 
</dd>
<dt><b>--out-counter-len</b><i>=Length</i>
</dt>
<dd> in bytes 
Length of counter field in output 
(default=4) 
</dd>
<dt><b>-C</b>,<b>--both-strands</b>
</dt>
<dd> Count both strand, canonical representation 
(default=off) 
</dd>
<dt><b>-p</b>,
<b>--reprobes</b><i>=INT</i>
</dt>
<dd> Maximum number of reprobes (default=62) 
</dd>
<dt><b>-r</b>,<b>--raw</b>
</dt>
<dd> Write raw database (default=off) 
</dd>
<dt><b>-q</b>,<b>--quake</b>
</dt>
<dd> Quake compatibility mode (default=off) 
</dd>
<dt><b>--quality-start</b><i>=INT</i>
</dt>
<dd> Starting ASCII for quality values 
(default=64) 
</dd>
<dt><b>--min-quality</b><i>=INT</i>
</dt>
<dd> Minimum quality. A base with lesser quality 
becomes an N (default=0) 
</dd>
<dt><b>-L</b>,
<b>--lower-count</b><i>=LONG</i>
</dt>
<dd> Don't output k-mer with count < lower-count 
</dd>
<dt><b>-U</b>,
<b>--upper-count</b><i>=LONG</i>
</dt>
<dd> Don't output k-mer with count > upper-count 
</dd>
<dt><b>--matrix</b><i>=Matrix</i>
</dt>
<dd> file Hash function binary matrix 
</dd>
<dt><b>--timing</b><i>=Timing</i>
</dt>
<dd> file Print timing information 
</dd>
</dl>
<h4><a name="section_5">histo</a></h4>

<p>
Create an histogram of k-mer occurrences 
<p>
Usage: jellyfish histo [OPTIONS]... [database.jf]... 
<p>
<dl compact>
<dt><b>--help</b>
</dt>
<dd> Print help and exit 
</dd>
<dt><b>-V</b>,<b>--version</b>
</dt>
<dd> Print version and exit 
</dd>
<dt><b>-s</b>,
<b>--buffer-size</b><i>=Buffer</i>
</dt>
<dd> length 
Length in bytes of input buffer 
(default=10000000) 
</dd>
<dt><b>-l</b>,
<b>--low</b><i>=LONG</i>
</dt>
<dd> Low count value of histogram (default=1) 
</dd>
<dt><b>-h</b>,
<b>--high</b><i>=LONG</i>
</dt>
<dd> High count value of histogram 
(default=10000) 
</dd>
<dt><b>-i</b>,
<b>--increment</b><i>=LONG</i>
</dt>
<dd> Increment value for buckets (default=1) 
</dd>
<dt><b>-t</b>,
<b>--threads</b><i>=INT</i>
</dt>
<dd> Number of threads (default=1) 
</dd>
<dt><b>-o</b>,
<b>--output</b><i>=STRING</i>
</dt>
<dd> Output file (default=/dev/fd/1) 
</dd>
</dl>
<h4><a name="section_6">dump</a></h4>

<p>
Dump k-mer counts 
<p>
Usage: jellyfish stats [OPTIONS]... [database.jf]... 
<p>
<dl compact>
<dt><b>-h</b>,<b>--help</b>
</dt>
<dd> Print help and exit 
</dd>
<dt><b>-V</b>,<b>--version</b>
</dt>
<dd> Print version and exit 
</dd>
<dt><b>-c</b>,<b>--column</b>
</dt>
<dd> Column format (default=off) 
</dd>
<dt><b>-t</b>,<b>--tab</b>
</dt>
<dd> Tab separator (default=off) 
</dd>
<dt><b>-L</b>,
<b>--lower-count</b><i>=LONG</i>
</dt>
<dd> Don't output k-mer with count < lower-count 
</dd>
<dt><b>-U</b>,
<b>--upper-count</b><i>=LONG</i>
</dt>
<dd> Don't output k-mer with count > upper-count 
</dd>
<dt><b>-o</b>,
<b>--output</b><i>=STRING</i>
</dt>
<dd> Output file (default=/dev/fd/1) 
</dd>
</dl>
<h4><a name="section_7">stats</a></h4>

<p>
Statistics 
<p>
Usage: jellyfish stats [OPTIONS]... [database.jf]... 
<p>
<dl compact>
<dt><b>-h</b>,<b>--help</b>
</dt>
<dd> Print help and exit 
</dd>
<dt><b>--full-help</b>
</dt>
<dd> Print help, including hidden options, and exit 
</dd>
<dt><b>-V</b>,<b>--version</b>
</dt>
<dd> Print version and exit 
</dd>
<dt><b>-L</b>,
<b>--lower-count</b><i>=LONG</i>
</dt>
<dd> Don't output k-mer with count < lower-count 
</dd>
<dt><b>-U</b>,
<b>--upper-count</b><i>=LONG</i>
</dt>
<dd> Don't output k-mer with count > upper-count 
</dd>
<dt><b>-v</b>,<b>--verbose</b>
</dt>
<dd> Verbose (default=off) 
</dd>
<dt><b>-o</b>,
<b>--output</b><i>=STRING</i>
</dt>
<dd> Output file (default=/dev/fd/1) 
</dd>
</dl>
<h4><a name="section_8">merge</a></h4>

<p>
Merge jellyfish databases 
<p>
Usage: jellyfish merge [OPTIONS]... [database.jf]... 
<p>
<dl compact>
<dt><b>-h</b>,<b>--help</b>
</dt>
<dd> Print help and exit 
</dd>
<dt><b>-V</b>,<b>--version</b>
</dt>
<dd> Print version and exit 
</dd>
<dt><b>-s</b>,
<b>--buffer-size</b><i>=Buffer</i>
</dt>
<dd> length 
Length in bytes of input buffer 
(default=10000000) 
</dd>
<dt><b>-o</b>,
<b>--output</b><i>=STRING</i>
</dt>
<dd> Output file (default=mer_counts_merged.jf) 
</dd>
<dt><b>--out-counter-len</b><i>=INT</i>
</dt>
<dd> Length (in bytes) of counting field in output 
(default=4) 
</dd>
<dt><b>--out-buffer-size</b><i>=LONG</i>
</dt>
<dd> Size of output buffer per thread 
(default=10000000) 
</dd>
<dt><b>-v</b>,<b>--verbose</b>
</dt>
<dd> Be verbose (default=off) 
</dd>
</dl>
<h4><a name="section_9">cite</a></h4>

<p>
How to cite Jellyfish's paper 
<p>
Usage: jellyfish cite [OPTIONS]... 
<p>
<dl compact>
<dt><b>-h</b>,<b>--help</b>
</dt>
<dd> Print help and exit 
</dd>
<dt><b>-V</b>,<b>--version</b>
</dt>
<dd> Print version and exit 
</dd>
<dt><b>-b</b>,<b>--bibtex</b>
</dt>
<dd> Bibtex format (default=off) 
</dd>
<dt><b>-o</b>,
<b>--output</b><i>=STRING</i>
</dt>
<dd> Output file (default=/dev/fd/1) 
</dd>
</dl>
<p>
<h2><a name="section_10">Version</a></h2>

<p>
Version: 1.1 of 2010/10/1
<p>
<h2><a name="section_11">Bugs</a></h2>

<p>
<ul compact>
<li><tt>jellyfish merge</tt> has not been parallelized and is very
slow. 
</li>
</ul>
<p>
<h2><a name="section_12">Copyright &amp; License</a></h2>

<dl compact>
<dt>Copyright</dt>
<dd> &copy; 2010, Guillaume Marcais <a href ="mailto:guillaume@marcais.net"><tt>guillaume@marcais.net</tt></a>
and Carl Kingsford <a href ="mailto:carlk@umiacs.umd.edu"><tt>carlk@umiacs.umd.edu</tt></a>.
<p>
</dd>
<dt>License</dt>
<dd> This program is free software: you can redistribute it 
and/or modify it under the terms of the GNU General Public License 
as published by the Free Software Foundation, either version 3 of 
the License, or (at your option) any later version. <br>

This program is distributed in the hope that it will be useful, but 
WITHOUT ANY WARRANTY; without even the implied warranty of 
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU 
General Public License for more details. <br>

You should have received a copy of the GNU General Public License 
along with this program. If not, see 
<<a href ="http://www.gnu.org/licenses/"><tt>http://www.gnu.org/licenses/</tt></a>>.
</dd>
</dl>
<p>
<h2><a name="section_13">Authors</a></h2>

Guillaume Marcais <br>

University of Maryland <br>

<a href ="mailto:gmarcais@umd.edu"><tt>gmarcais@umd.edu</tt></a>
<p>
Carl Kingsford <br>

University of Maryland <br>

<a href ="mailto:carlk@umiacs.umd.edu"><tt>carlk@umiacs.umd.edu</tt></a>
<p>
</body>
</html>
<!-- NOTE: This file is generated, DO NOT EDIT. -->
